### ISSUE TITLE
jpmignore doesn't exclude manifest.json

### ISSUE THREAD

**Issue by Leoperd87** [mozilla-jetpack/jpm#564]
.jpmignore:
```
*.ico
manifest.json
```

```
...
JPM [error] XPI creation failed: Error: Both "package.json" and "manifest.json" files found in the add-on root dir.
...
```

**Comment by freaktechnik** [comment]
Does your `.jpmignore` have a newline between the two entries? Because rules need to be separated by newlines, afaik.

**Comment by Leoperd87** [comment]
> Does your .jpmignore have a newline between the two entries?
Yes.

**Comment by kumar303** [comment]
manifest.json is a special file because it contains metadata. This will conflict with package.json, which also contains metadata.

I'm slightly confused. Are you building a WebExtension or an SDK add-on that happens to have a manifest.json for some other reason? If you are building a WebExtension, use the [web-ext](https://github.com/mozilla/web-ext) tool instead. It has more features than jpm ;)

**Comment by Leoperd87** [comment]
Many years we use same sourcecode for IE, Safari, FF and Chrome extensions.
Source folder contain Icon.png, manifest.json, READ.ME, data, Info.plist, package.json, update-manifest.plist...

manifest.json - Chrome extension data
Info.plist - Safari extension data
package.json - FF extension data

I'm slightly confused. If I told jpm to ignore manifest.json (Chrome extension data!!!!), why it show me error message.

**Comment by freaktechnik** [comment]
> I'm slightly confused. If I told jpm to ignore manifest.json (Chrome extension data!!!!), why it show me error message.

Because a manifest.json would be parsed before packaging, as the .jpmignore only affects packaging.

**Comment by Leoperd87** [comment]
Do you mean that you want to parse manifest of another browser, just because you can, and everything other was developer's problem?

Sorry.

Any way, if you want to support chrome BX you can do it.
I think it's realy strange:

- Parse manifest.json without any special jpm parameters.
- Parse manifest.json if package.json exist.

**Comment by kumar303** [comment]
I think I understand. Yes, Firefox does support Chrome extensions now :) The problem is that you can't mix a WebExtension (the new manifest.json standard) with an older SDK add-on. I guess we could patch JPM to fix this. 

One workaround would be to move your SDK add-on source into a new directory that does not have a manifest.json in it. I don't know if that will be easy or hard. You could also start using the [web-ext](https://github.com/mozilla/web-ext) tool to ship a WebExtension but it would only support Firefox 48 and higher.

**Comment by Leoperd87** [comment]
Thank you!

Kind regards,
Leoperd87

**Comment by kumar303** [comment]
Hi. Sorry about the inconvenience here. I looked into this deeper and I think it's a regression. I filed https://github.com/mozilla-jetpack/jpm/issues/565 to fix it. The quickest workaround for you would be to downgrade JPM: `npm install jpm@1.1.2`. That will not give you an error.

Since I created a new bug that addresses the root cause better, I'm closing this bug. Thanks for the report!

**Comment by kumar303** [comment]
@Leoperd87 I'm slightly confused that this worked for you for so long because I thought addons.mozilla.org would refuse to sign an add-on with both `install.rdf` and `manifest.json`. Have you been distributing your add-on on AMO successfully in recent months?

**Comment by kumar303** [comment]
@Leoperd87 btw, we just released 1.1.4 of jpm that fixes the `manifest.json` error. https://www.npmjs.com/package/jpm
